### ISSUE TITLE
BoltztrapRunner discards bands with no warning

### ISSUE THREAD

**Issue by utf** [materialsproject/pymatgen#842]
The `BoltztrapRunner` class discards the top 10% of bands when writing the eigenvalues to the input files needed by BoltzTraP. In some cases this can cause the valence band maximum and conduction band minimum to be discarded, leading to BoltzTraP errors. I understand the rationale for discarding these bands (i.e. that often they are not very well converged) but I think it is presumptuous to discard them. For example, calculations performed using different codes, or with `ALGO = Exact` in VASP, will not suffer from the same problems. 

## Example code

On lines 254-257 of `pymatgen/electronic_structure/boltztrap.py`:
```python
# use 90% of bottom bands since highest eigenvalues
# are usually incorrect
# ask Geoffroy Hautier for more details
nb_bands = int(math.floor(self._bs.nb_bands * 0.9))
```

## Error message

If the valence band maximum and conduction band minimum are both discarded, the `BANDANA` section of the `boltztrap.outputtrans` file will read:

```
==============  OUTPUT from BANDANA       ================
Egap:     0.000000 Energy range:    -0.176310 -    -0.049760. Bands range:   47 -   57
VBM:     0.000000 CBM:     0.000000 Efermi:     0.000000
```

BoltzTraP will also warn the user with the message:

```
Error - Fermi level was not found for doping
```

## Suggested solution

Either remove this code, or check to see if the bands removed include the VBM/CBM+2/3, in which case don't scrap these bands. At the minimum a warning should be thrown.

**Comment by shyuep** [comment]
I agree, and I would say a parameter should be set to allow people to choose to discard a certain % or none at all. I will leave @hautierg and @computron to comment and implement.

**Comment by computron** [comment]
I think the relevant person here is @fraricci to implement and @hautierg to comment on the procedure (as per the explicit comment that Geoffroy suggests throwing out the top 10% of bands)

**Comment by fraricci** [comment]
I will implement what @shyuep  suggested: a parameter to choose the % of the band cut, with 10% as default.

**Comment by utf** [comment]
This was fixed by #850
